bundles/framework/featuredata2/plugin/FeaturedataPlugin.js

```diff
@@ -221,6 +221,20 @@ Oskari.clazz.define('Oskari.mapframework.bundle.featuredata2.plugin.FeaturedataP
                  */
                 AfterMapMoveEvent: function (event) {
                     this.refresh();
+                },
+                /**
+                 * @method AfterMapLayerAddEvent
+                 * @param {Oskari.mapframework.event.common.AfterMapLayerAddEvent} event
+                 *
+                 * Calls flyouts layerAdded() method
+                 */
+                'AfterMapLayerAddEvent': function (event) {
+                    if (event.getMapLayer().hasFeatureData()) {
+                        this.refresh();
+                    }
+                },
+                'AfterMapLayerRemoveEvent': function (event) {
+                    this.refresh();
```

Comment:
Do we need to refresh here if layer doesn't have featuredata? Or could we add the same if here as in add?